### PR TITLE
CASSANDRA-18729 5.0 cleanup of Netty upgrade

### PR DESCRIPTION
**Pull request by smiklosovic**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: .build/parent-pom-template.xml [hunk 1]
```diff
@@ -747,14 +813,20 @@
       <dependency>
         <groupId>io.netty</groupId>
         <artifactId>netty-transport-native-epoll</artifactId>
-        <version>4.1.94.Final</version>
+        <version>4.1.96.Final</version>
```
Comment (jacek-lewandowski) [inline .build/parent-pom-template.xml:809]: can you define netty version in a property? (no need to rerun the tests)

Comment (smiklosovic) [reply .build/parent-pom-template.xml:809]: @jacek-lewandowski we will dedicate a separate ticket for this, I just do not want to have it inconsistent, all versions should go to properties IMHO.

Comment (jacek-lewandowski) [reply .build/parent-pom-template.xml:809]: ok

File: test/distributed/org/apache/cassandra/distributed/test/TestBaseImpl.java [hunk 2]
```diff
@@ -61,6 +62,7 @@ public void afterEach() {
     public static void beforeClass() throws Throwable
     {
         ICluster.setup();
+        CassandraRelevantProperties.IO_NETTY_TRANSPORT_NO_NATIVE.setBoolean(false);
```
Comment (jacek-lewandowski) [inline test/distributed/org/apache/cassandra/distributed/test/TestBaseImpl.java]: So, alternatively you may set:
```
CassandraRelevantProperties.NATIVE_EPOLL_ENABLED.setBoolean(false);
```
which will be consistent with that settings in `ICluster`. Also, perhaps a better place to set this or that option is initialization block in `AbstractCluster`:

```
        {
            // Indicate that we are running in the in-jvm dtest environment
            CassandraRelevantProperties.DTEST_IS_IN_JVM_DTEST.setBoolean(true);
            // those properties may be set for unit-test optimizations; those should not be used when running dtests
            CassandraRelevantProperties.TEST_FLUSH_LOCAL_SCHEMA_CHANGES.reset();
            CassandraRelevantProperties.NON_GRACEFUL_SHUTDOWN.reset();
...
```

However, there also another problem reported when running those tests:

```
org.hyperic.sigar.SigarException: Native Library /home/jlewandowski/dev/cassandra/trunk/lib/sigar-bin/libsigar-amd64-linux.so already loaded in another classloader
	at org.hyperic.sigar.Sigar.loadLibrary(Sigar.java:172)
	at org.hyperic.sigar.Sigar.<clinit>(Sigar.java:100)
	at org.apache.cassandra.utils.SigarLibrary.<init>(SigarLibrary.java:49)
	at org.apache.cassandra.utils.SigarLibrary.<clinit>(SigarLibrary.java:30)
	at org.apache.cassandra.service.StartupChecks$8.execute(StartupChecks.java:367)
	at org.apache.cassandra.service.StartupChecks.verify(StartupChecks.java:175)
	at org.apache.cassandra.service.CassandraDaemon.runStartupChecks(CassandraDaemon.java:495)
	at org.apache.cassandra.distributed.impl.Instance.lambda$startup$12(Instance.java:620)
	at org.apache.cassandra.concurrent.FutureTask$1.call(FutureTask.java:96)
	at org.apache.cassandra.concurrent.FutureTask.call(FutureTask.java:61)
	at org.apache.cassandra.concurrent.FutureTask.run(FutureTask.java:71)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

which makes all Sigar based stuff unused in some instances. I think we should mark `SigarLibrary` class with `Shared` annotation to fix that.

**Review by jacek-lewandowski — APPROVED** [review]
I've left one comment, other than that I'm +1 with green tests

**Pull request closed without merging**